src/retasc/run.py

```diff
@@ -82,7 +86,16 @@ def run(*, config: Config, jira_token: str, dry_run: bool) -> Report:
             context.template.params["jira_issues"] = context.report.jira_issues
             update_state(rule, context)
 
-    if dry_run:
-        logger.warning("To apply changes, run without --dry-run flag")
+    return report
+
+
+@tracer.start_as_current_span("run")
+def run(*, config: Config, jira_token: str) -> Report:
+    return run_helper(config=config, jira_token=jira_token, jira_cls=JiraClient)
+
 
+@tracer.start_as_current_span("dry_run")
+def dry_run(*, config: Config, jira_token: str) -> Report:
+    report = run_helper(config=config, jira_token=jira_token, jira_cls=DryRunJiraClient)
```

Comment:
Should we add some checks on the DryRunJiraClient to make sure all the edit functions has the dry-run function? 
It's about forbidding any future edit functions added and forgot a dry-run function.

Reply:
That is an interesting idea, but I'm not sure if it can be easily tested.